extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -49,21 +47,32 @@ export function BaseCarousel({
     carouselContext.setCurrentSlide ?? setCurrentSlideState;
   const {setSlideCount} = carouselContext;
   const scrollRef = useRef(null);
-  const advance = (by) => scrollRef.current.advance(by);
-  useMountEffect(() => {
-    if (setAdvance) {
-      setAdvance(advance);
-    }
-  });
+
+  const advance = useCallback((by) => scrollRef.current.advance(by), []);
+  const setRestingIndex = useCallback(
+    (i) => {
+      setCurrentSlide(i);
+      if (onSlideChange) {
+        onSlideChange(i);
+      }
+    },
+    [setCurrentSlide, onSlideChange]
+  );
+
+  useImperativeHandle(
+    ref,
+    () =>
+      /** @type {!BaseCarouselDef.CarouselApi} */ ({
+        advance,
+        goToSlide: (index) => setRestingIndex(index),
```

Comment:
Since this is exposed, let's either assert that `index < children.length` or use `index % children.length` by default.

Reply:
Resolved using `clamp`